### PR TITLE
Various PHP 8 fixes

### PR DESCRIPTION
**Pull request by aschempp**

### CODE REVIEW

File: core-bundle/src/Translation/MessageCatalogue.php [hunk 1]
```diff
@@ -178,6 +178,10 @@ private function getFromGlobals(string $id): ?string
             $item = &$item[$part];
         }
 
-        return $item;
+        if (\is_array($item)) {
+            return $id;
```
Comment (ausi) [inline core-bundle/src/Translation/MessageCatalogue.php]:
```suggestion
            return null;
```

File: core-bundle/src/Resources/contao/drivers/DC_Table.php [hunk 2]
```diff
@@ -4761,7 +4761,7 @@ protected function parentView()
 	protected function listView()
 	{
 		$table = ($GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['mode'] ?? null) == self::MODE_TREE_EXTENDED ? $this->ptable : $this->strTable;
-		$orderBy = $GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['fields'] ?? array();
+		$orderBy = $GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['fields'] ?? array('id');
 		$firstOrderBy = preg_replace('/\s+.*$/', '', $orderBy[0]);
```
Comment (ausi) [inline core-bundle/src/Resources/contao/drivers/DC_Table.php:4765]: Shouldn’t we fix this on the second line instead?

```suggestion
		$orderBy = $GLOBALS['TL_DCA'][$this->strTable]['list']['sorting']['fields'] ?? array();
		$firstOrderBy = preg_replace('/\s+.*$/', '', $orderBy[0] ?? '');
```

Comment (aschempp) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:4765]: I thought about this as well, but ordering by an empty string is imho worse than ordering by the record id (maybe sorting by empty won't even work).

Comment (ausi) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:4765]: I’m not unserstanding all the rest of the code, is it guaranteed that there is always a column named `'id'`?
If so, using `array('id')` might be better I agree.

Comment (aschempp) [reply core-bundle/src/Resources/contao/drivers/DC_Table.php:4765]: well it's a DCA, and our DCA does not work without an ID column imho 🤷

**Review by ausi — DISMISSED** [review]

File: core-bundle/src/Resources/contao/library/Contao/Controller.php [hunk 3]
```diff
@@ -211,7 +211,7 @@ public static function getTemplateGroup($strPrefix, array $arrAdditionalMapper=a
 					}
 				}
 
-				$arrTemplates[$strTemplate][] = $GLOBALS['TL_LANG']['MSC']['global'];
+				$arrTemplates[$strTemplate][] = $GLOBALS['TL_LANG']['MSC']['global'] ?? 'Global';
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/library/Contao/Controller.php]:
```suggestion
				$arrTemplates[$strTemplate][] = $GLOBALS['TL_LANG']['MSC']['global'] ?? 'global';
```

**Comment by leofeyer** [comment]
What was the issue in the `MessageCatalogue` class?

**Comment by aschempp** [comment]
> What was the issue in the `MessageCatalogue` class?

```
$GLOBALS['TL_LANG']['MSC']['foo']['bar'] = 'baz';

$this->trans('MSC.foo')
```

will lead to an array to string casting error.

File: core-bundle/src/Resources/contao/library/Contao/Controller.php (hunk 3 above)
Comment (aschempp) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php]: why? It's the title of the global template `optgroup`?

Comment (leofeyer) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php]: The default string for `$GLOBALS['TL_LANG']['MSC']['global']` is also `global`, therefore I figured it should be consistent. What do you mean by _global template optgroup_?

**Comment by leofeyer** [comment]
The same error would be thrown if you did `echo $GLOBALS['TL_LANG']['MSC']['foo']`. I don‘t think we should ignore it, so the developer realizes that something is wrong.

**Review by leofeyer — APPROVED** [review]

**Pull request merged**